### PR TITLE
feat(snapshot-clone): csi-controller to allow creation from snapshot

### PR DESCRIPTION
**Pull request by Abhinandan-Purkait**

### CODE REVIEW

File: control-plane/csi-driver/src/bin/controller/controller.rs [hunk 1]
```diff
@@ -178,11 +178,44 @@ impl rpc::csi::controller_server::Controller for CsiControllerSvc {
...
+                        .await?
+                        .state
+                        .ready_as_source
+                    {
+                        Some(snapshot_uuid)
+                    } else {
+                        return Err(Status::failed_precondition(
+                            format!("Volume creation from snapshot source is not allowed for {snapshot_uuid}"),
+                        ));
+                    }
```
Comment (tiagolobocastro) [inline control-plane/csi-driver/src/bin/controller/controller.rs]: Shouldn't need to do this, create_volume will validate this

**Review by tiagolobocastro — APPROVED** [review]

File: control-plane/csi-driver/src/bin/controller/controller.rs (hunk 1 above)
Comment (Abhinandan-Purkait) [reply control-plane/csi-driver/src/bin/controller/controller.rs]: Cool. Removing this then.

**Comment by Abhinandan-Purkait** [comment]
bors merge

**Comment by Abhinandan-Purkait** [comment]
bors merge

**Pull request merged**